src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -272,7 +272,8 @@ internal BoundExpression ConvertPatternExpression(TypeSymbol inputType, CSharpSy
                     hasErrors = CheckRestrictedTypeInAsync(this.ContainingMemberOrLambda, declType, diagnostics, typeSyntax);
                 }
 
-                return new BoundDeclarationPattern(node, localSymbol, boundDeclType, isVar, hasErrors);
+                var expression = localSymbol == null ? new BoundDiscardExpression(node, boundDeclType.Type) : (BoundExpression)new BoundLocal(node, localSymbol, null, boundDeclType.Type);
```

Comment:
not understanding why it's the case that we have a discard if we don't have a local symbol.